src/permission/fs_permission.cc

```diff
@@ -20,13 +20,14 @@ std::string WildcardIfDir(const std::string& res) noexcept {
   uv_fs_t req;
   int rc = uv_fs_stat(nullptr, &req, res.c_str(), nullptr);
   if (rc == 0) {
-    const uv_stat_t* const s = static_cast<const uv_stat_t*>(req.ptr);
+    const auto s = static_cast<const uv_stat_t*>(req.ptr);
     if ((s->st_mode & S_IFMT) == S_IFDIR) {
       // add wildcard when directory
-      if (res.back() == node::kPathSeparator) {
+      if (res.back() == std::filesystem::path::preferred_separator) {
         return res + "*";
       }
-      return res + node::kPathSeparator + "*";
+
+      return (std::filesystem::path(res) / "*").string();
```

Comment:
I didn't test it, but from `std::filesystem::path::operator/=` it might cause an unexpected replacement:

```
// Where "//host" is a root-name
path("//host")  / "foo" // the result is      "//host/foo" (appends with separator)
path("//host/") / "foo" // the result is also "//host/foo" (appends without separator)
 
// On POSIX,
path("foo") / ""      // the result is "foo/" (appends)
path("foo") / "/bar"; // the result is "/bar" (replaces)
```

Refs: https://en.cppreference.com/w/cpp/filesystem/path/append.

Possibly, the "*" append won't cause any error, but I would be more confident if we use `path::preferred_separator` here.

Reply:
I can replace it, but FYI, we can use `concat` and it will always work correctly.

```cpp
#include <iostream>
#include <filesystem>

int main() {
  auto path = std::filesystem::path("foo").concat("/bar");
  printf("path -> %s\n", path.string().c_str());
  return 0;
}
```

Prints:
```
path -> foo/bar
```

Reply:
I suspect it would cause more CPU cycles though. IMO the `+` concat is easier

Reply:
Agreed. I'm changing this. Thank you for the review.

Reply:
Unfortunately there has been a complication. On Windows, preferred_separator returns "wchar_t*". Therefore, it is not possible to make a simple addition of wchar_t and std::string.

The solution to fix this is:

```c++
#ifdef _WIN32
      // This is required for Windows since prefered separator is "wchar_t".
      auto separator = std::filesystem::path::preferred_separator;
      return res + std::string(separator.begin(), separator.end()) + "*";
#else
      return res + std::filesystem::path::preferred_separator + "*";
#endif
    }
```

But since we are talking about potential solutions, I thought what about just converting this whole function to use std::filesystem, rather than libuv, and make the whole code a 5 line function?

```c++
std::string WildcardIfDir(const std::string& res) noexcept {
  auto path = std::filesystem::path(res);
  auto file_status = std::filesystem::status(path);
  if (file_status.type() == std::filesystem::file_type::directory) {
      return (path / "*").string();
  }
  return path.string();
}
```

I personally prefer the std::filesystem, but wanted to hear @RafaelGSS @tniessen and @jasnell's thoughts on this.

Reply:
For now I think the first change is better. I think we'll want to carefully consider switching too much to `std::filesystem` until we can be certain it's implemented consistently across all the platforms we support (unless you're thinking of using this variation only on Windows)

Reply:
I would not change it to the former either as `std::filesystem` is very likely to perform syscalls and the Permission Model operates on non-existing paths and it would be more difficult to handle. That's why I used `std::string_view`